policy/predicate/file.go

```diff
@@ -181,6 +181,60 @@ func (pred *NoChangedFiles) Trigger() common.Trigger {
 	return common.TriggerCommit
 }
 
+type FileNotDeleted struct {
+	Paths []common.Regexp `yaml:"paths"`
+}
+
+var _ Predicate = &FileNotDeleted{}
+
+func (pred *FileNotDeleted) Evaluate(ctx context.Context, prctx pull.Context) (*common.PredicateResult, error) {
+	var paths []string
+	for _, r := range pred.Paths {
+		paths = append(paths, r.String())
+	}
+
+	predicateResult := common.PredicateResult{
+		Satisfied:       true,
+		ValuePhrase:     "file not deleted",
+		Values:          []string{},
+		ConditionPhrase: "all specified paths exist",
+		ConditionValues: paths,
+	}
+
+	changedFiles, err := prctx.ChangedFiles()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list changed files")
+	}
+
+	if len(changedFiles) == 0 {
+		predicateResult.Description = "No files were changed"
+		return &predicateResult, nil
+	}
+
+	var deletedFiles []string
+	for _, f := range changedFiles {
+		if anyMatches(pred.Paths, f.Filename) {
+			if f.Status == pull.FileDeleted {
```

Comment:
```suggestion
		if f.Status == pull.FileDeleted {
			if anyMatches(pred.Paths, f.Filename) {
```

Let's check the file status first to avoid doing any regex evaluation for files that we already know have no impact.

Reply:
Good catch, thanks!